### ISSUE TITLE
Can't add same device again after removing it

### ISSUE THREAD

**Issue by admiralalec** [rospogrigio/localtuya#933]
## The problem
I added a device, forgot a senser, removed it and tried re-adding again. 
It results in the an error directing me to the log. 

I have this with 2 devices. 

I don't use the cloud connection - so directly connect with local_key from Tuya API. 

## Environment

- Localtuya version: v4.0.1
- Home Assistant Core version: core-2022.6.7
- [] Are you using the Home Assistant Tuya Cloud component ? 
- [] Are you using the Tuya App in parallel ? 

## Steps to reproduce

1. Add device 
2. Remove device 
3. Try to re-add 

## Provide Home Assistant taceback/logs
```
Unexpected exception
Traceback (most recent call last):
  File "/config/custom_components/localtuya/config_flow.py", line 533, in async_step_configure_device
    self.dps_strings = await validate_input(self.hass, user_input)
  File "/config/custom_components/localtuya/config_flow.py", line 235, in validate_input
    interface = await pytuya.connect(
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 669, in connect
    _, protocol = await loop.create_connection(
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1065, in create_connection
    raise exceptions[0]
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1050, in create_connection
    sock = await self._connect_sock(
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 961, in _connect_sock
    await self.sock_connect(sock, address)
  File "/usr/local/lib/python3.9/asyncio/selector_events.py", line 499, in sock_connect
    return await fut
  File "/usr/local/lib/python3.9/asyncio/selector_events.py", line 534, in _sock_connect_cb
    raise OSError(err, f'Connect call failed {address}')
OSError: [Errno 113] Connect call failed ('192.168.20.135', 6668)
```

**Comment by balbulator** [comment]
Same. Two identical devices (sockets), identical firmware, work perfectly in the phone through SmartLife. One of them connects perfectly in LocalTuya, and the other gives an error when connecting. The problem started approximately with the transition to LocalTuya v.4.0

**Comment by balbulator** [comment]
Physically reconnecting this device (unplugging it) helped me. After that, it connected to LocalTuya

**Comment by admiralalec** [comment]
Didn't work for me... 

> Physically reconnecting this device (unplugging it) helped me. After that, it connected to LocalTuya

**Comment by CloCkWeRX** [comment]
Duplicate of #897

**Comment by CloCkWeRX** [comment]
> OSError: [Errno 113] Connect call failed ('192.168.0.35', 6668)

To translate this one, your device can't connect to that IP / Port.
I suggest you check the device is actually up and running with a tool like nmap.

The likely reason that powercycling it (unplug/replug) fixed it for some is probably causing the device to start actually listening.

The bug here; you should get a connection failed shown in the UI.

**Comment by Greatdane** [comment]
@CloCkWeRX I get the same error above, but the plug continues to works in the smart life app and the official tuya home assistant integration, so I don't think its a connection error.

Plus, a lot of people seem to have the same issue recently.

**Comment by RACMan2527** [comment]
Indeed, same issus as [CloCkWeRX]

I get the same error above, but the plug AND the energy reading continue to works in the smart life app and the official tuya home assistant integration, so I don't think its a connection error.

I have four of the same plug.  Two are continuing to work while two went 'unavailable' in HA (still work in SmartLife app).
Removed the two 'unavailable' from HA localTuya (version 5.0.0).  Cannot reinstall.
Upgraded to  localTuya (version 5.1.0).  No change.  Still working in SmartLife app.

I am unable to remove/replace the plug from the wall without disrupting equipment attached.
